Modules/Core/Common/include/itkMacro.h

```diff
@@ -99,18 +99,27 @@ namespace itk
 // to be quoted.
 #define ITK_PRAGMA(x) _Pragma(#x)
 
-// The clang compiler has many useful non-default compiler warnings
-// that tend to have a high false positive rate.
-// The following set of defines allows us to suppress false positives
-// and still track down suspicious code
+// The GCC/Clang compilers have many useful non-default compiler warnings
+// that tend to have a high false positive rate or are otherwise not always appropriate.
+// The following set of defines allows us to suppress instances of said warnings.
+
+// For GCC and Clang (Clang also identifies itself as GCC, and supports these pragmas):
+#if defined(__GNUC__)
+#  define GCC_PRAGMA_PUSH ITK_PRAGMA(GCC diagnostic push)
+#  define GCC_PRAGMA_POP ITK_PRAGMA(GCC diagnostic pop)
+#  define GCC_SUPPRESS_Wfloat_equal ITK_PRAGMA(GCC diagnostic ignored "-Wfloat-equal")
+#  define GCC_SUPPRESS_Wformat_nonliteral ITK_PRAGMA(GCC diagnostic ignored "-Wformat-nonliteral")
+#else
+#  define GCC_PRAGMA_PUSH
+#  define GCC_PRAGMA_POP
+#  define GCC_SUPPRESS_Wfloat_equal
+#  define GCC_SUPPRESS_Wformat_nonliteral
```

Comment:
Can you please add "ITK" as prefix to these macro names? It appears that macro's like these are also accessible in user code, so it should be made clear that they are introduced by ITK, not by GCC. (I see, we did forget the "ITK" prefix before, but this PR of yours is a nice opportunity to do it right 😇 !)

Reply:
Thanks @hjmjohnson, cool  👍